crev-lib/src/local.rs

```diff
@@ -985,18 +1033,43 @@ impl ProofStore for Local {
 
     fn proofs_iter(&self) -> Result<Box<dyn Iterator<Item = proof::Proof>>> {
         Ok(Box::new(
-            self.get_proofs_dir_path_opt()?
-                .into_iter()
-                .flat_map(proofs_iter_for_path),
+            self.proofs_for_current_url()?.map(|(proof, _)| proof)
         ))
     }
 }
 
-fn proofs_iter_for_path(path: PathBuf) -> impl Iterator<Item = proof::Proof> {
+/// Scan a directory of git checkouts. Assumes fetch source is the origin URL.
+fn proofs_iter_for_remotes_checkouts(path: PathBuf) -> Result<impl Iterator<Item = (proof::Proof, FetchSource)>>{
+    let dir = std::fs::read_dir(&path)?;
+    Ok(dir
+        .filter_map(|e| e.ok())
+        .filter_map(|e| {
+            if let Ok(ty) = e.file_type() {
+                if ty.is_dir() {
+                    return Some(e.path());
+                }
+            }
+            None
+        })
+        .filter_map(move |path| {
+            let repo = git2::Repository::open(&path).ok()?;
+            let origin = repo.find_remote("origin").ok()?;
+            let fetch_source = FetchSource::Url(Arc::new(Url::new_git(origin.url()?)));
+            Some(proofs_iter_for_path(path, fetch_source))
+        })
+        .flat_map(|iter| iter)
+    )
+}
+
+/// Scan a git checkout or any subdirectory obtained from a known URL
+fn proofs_iter_for_path(path: PathBuf, fetch_source: FetchSource) -> impl Iterator<Item = (proof::Proof, FetchSource)> {
     use std::ffi::OsStr;
-    let file_iter = walkdir::WalkDir::new(path)
+
+    let file_iter = walkdir::WalkDir::new(&path)
         .into_iter()
-        .map_err(|e| format_err!("Error iterating local ProofStore: {:?}", e))
+        // skip dotfiles, .git dir
+        .filter_entry(|e| e.file_name().to_str().map_or(true, |f| !f.starts_with('.')))
```

Comment:
`.git` is definitely a good improvement, but are dotfiles worthwhile? Could it confuse anyone?

Reply:
The repo layout is maintained by crev, so I don't think it would be a problem. Even in manually-crafted repo users can figure out that dot-dirs are special.